syntax/asciidoctor.vim

```diff
@@ -63,27 +63,34 @@ syn sync match syncH4 grouphere NONE "^=====\s.*$"
-" contain a link/image/url/xref/mailto/etc, its syntax is
-" type::\?url[description], where description cannot be empty
-
-syn match asciidoctorFile  "\f\+" contained
-syn match asciidoctorMacro "\l\+::\?\S*\[.\{-}\]"  contains=asciidoctorUrl,asciidoctorLink
-syn match asciidoctorMacro "\s*\l\+://\S\+" contains=asciidoctorUrlAuto
 syn match asciidoctorAttribute "{[[:alpha:]][[:alnum:]-_:]\{-}}" 
 
+" a Macro is a generic pattern that has no default highlight, but it could contain a link/image/url/xref/mailto/etc, and naked URLs as well
+
+syn match asciidoctorMacro "\<\l\{-1,}://\S\+" contains=asciidoctorUrlAuto
+syn match asciidoctorMacro "\<\w.\{-}@\w\+\.\w\+" contains=asciidoctorEmailAuto
+syn match asciidoctorMacro "\<\l\{-1,}::\?\S*\[.\{-}\]"  contains=asciidoctorUrl,asciidoctorLink,asciidoctorEmail
+
+syn match asciidoctorFile           "\f\+" contained
 syn match asciidoctorUrlDescription "\[[^]]\{-}\]\%(\s\|$\)" contained containedin=asciidoctorLink
-syn match asciidoctorUrlAuto "\s*\zs\%(http\|ftp\)s\?://\S\+\%(\[.\{-}\]\)\?" contained contains=asciidoctorUrl
+syn match asciidoctorUrlAuto        "\%(http\|ftp\|irc\)s\?://\S\+\%(\[.\{-}\]\)\?" contained contains=asciidoctorUrl
+syn match asciidoctorEmailAuto      "[a-zA-Z0-9._%+-]\{-1,}@\w\+\.\w\+" contained
 
 if get(g:, 'asciidoctor_syntax_conceal', 0)
-	" the pattern \[\ze\%(\s*[^ ]\+\s*\)\+]\+ means: a brackets pair, inside of
-	" which at least one non-space character, possibly with spaces
-	syn region asciidoctorLink       matchgroup=Conceal start="\%(link\|xref\|mailto\):\ze[^:].*" end="\ze\[\s*\]" concealends oneline keepend skipwhite contained nextgroup=asciidoctorUrlDescription contains=asciidoctorUrl,asciidoctorFile
-	syn region asciidoctorLink       matchgroup=Conceal start="\%(video\|image\)::\?\ze.*" end="\ze\[.*\]" concealends oneline keepend skipwhite contained nextgroup=asciidoctorUrlDescription contains=asciidoctorFile
-	syn region asciidoctorLink       matchgroup=Conceal start="\%(link\|xref\|mailto\):[^:].*\[\ze\%(\s*[^ ]\+\s*\)\+]\+" end="\]" concealends oneline keepend skipwhite contained
-	" syn region asciidoctorLink       matchgroup=Conceal start="\%(video\|image\)::.*\[\ze\%(\s*[^ ]\+\s*\)\+]\+" end="\]" concealends oneline keepend skipwhite contained
+	" the pattern \[\ze\%(\s*[^ ]\+\s*\)\+]\+ means: a brackets pair, inside of which at least one non-space character, possibly with spaces
+	syn region asciidoctorLink       matchgroup=Conceal start="\%(link\|xref\|mailto\|irc\):\ze[^:].*" end="\ze\[\s*\]" concealends oneline keepend skipwhite contained nextgroup=asciidoctorUrlDescription contains=asciidoctorUrl,asciidoctorFile
+	syn region asciidoctorLink       matchgroup=Conceal start="\%(link\|xref\|mailto\|irc\):[^:].*\[\ze\%(\s*[^ ]\+\s*\)\+]\+" end="\]" concealends oneline keepend skipwhite contained
+
+	" conceal also the address of an image/video, if the description is not empty
+	if get(g:, 'asciidoctor_compact_media_links', 0)
```

Comment:
This option is undocumented, it relates to the alternatives I talked about [here](https://github.com/habamax/vim-asciidoctor/issues/47#issuecomment-576830701). It defaults to the current behaviour of master (verbose variant).

It's here so that if you don't want an option or alternative variants for this, it's easy to pick one by deleting the other one.

Reply:
I will keep it the way you did it.